holidays/countries/germany.py

```diff
@@ -102,6 +119,1200 @@ def __init__(self, *args, **kwargs) -> None:
+            while current_date <= end_date:
+                if current_date.year == self._year:
+                    self._add_holiday(name, current_date.month, current_date.day)
+                current_date += timedelta(days=1)
+
+        holidays_map = []
+
+        if year == 2025:
+            holidays_map = [
+                (tr("Weihnachtsferien"), date(2025, 12, 22), date(2025, 12, 31)),
+            ]
+        elif year == 2026:
+            holidays_map = [
+                (tr("Weihnachtsferien"), date(2026, 1, 1), date(2026, 1, 5)),
+                (tr("Halbjahresferien"), date(2026, 2, 2), date(2026, 2, 3)),
+                (tr("Osterferien"), date(2026, 3, 23), date(2026, 4, 7)),
+                (tr("Tag nach Himmelfahrt"), date(2026, 5, 15), date(2026, 5, 15)),
+                (tr("Pfingsten"), date(2026, 5, 26), date(2026, 5, 26)),
+                (tr("Sommerferien"), date(2026, 7, 2), date(2026, 8, 12)),
+                (tr("Herbstferien"), date(2026, 10, 12), date(2026, 10, 24)),
+                (tr("Weihnachtsferien"), date(2026, 12, 23), date(2026, 12, 31)),
+            ]
+        elif year == 2027:
+            holidays_map = [
+                (tr("Weihnachtsferien"), date(2027, 1, 1), date(2027, 1, 9)),
+                (tr("Halbjahresferien"), date(2027, 2, 1), date(2027, 2, 2)),
+                (tr("Osterferien"), date(2027, 3, 22), date(2027, 4, 3)),
+                (tr("Tag nach Himmelfahrt"), date(2027, 5, 7), date(2027, 5, 7)),
+                (tr("Pfingsten"), date(2027, 5, 18), date(2027, 5, 18)),
+                (tr("Sommerferien"), date(2027, 7, 8), date(2027, 8, 18)),
+                (tr("Herbstferien"), date(2027, 10, 18), date(2027, 10, 30)),
+                (tr("Weihnachtsferien"), date(2027, 12, 23), date(2027, 12, 31)),
+            ]
+        elif year == 2028:
+            holidays_map = [
+                (tr("Weihnachtsferien"), date(2028, 1, 1), date(2028, 1, 8)),
+                (tr("Halbjahresferien"), date(2028, 1, 31), date(2028, 2, 1)),
+                (tr("Osterferien"), date(2028, 4, 10), date(2028, 4, 22)),
+                (tr("Tag nach Himmelfahrt"), date(2028, 5, 26), date(2028, 5, 26)),
+                (tr("Pfingsten"), date(2028, 6, 6), date(2028, 6, 6)),
+                (tr("Sommerferien"), date(2028, 7, 20), date(2028, 8, 30)),
+                (tr("Tag vor dem 03. Oktober"), date(2028, 10, 2), date(2028, 10, 2)),
+                (tr("Herbstferien"), date(2028, 10, 23), date(2028, 11, 4)),
+                (tr("Weihnachtsferien"), date(2028, 12, 27), date(2028, 12, 31)),
+            ]
+        elif year == 2029:
+            holidays_map = [
+                (tr("Weihnachtsferien"), date(2029, 1, 1), date(2029, 1, 6)),
+                (tr("Halbjahresferien"), date(2029, 2, 1), date(2029, 2, 2)),
+                (tr("Osterferien"), date(2029, 3, 19), date(2029, 4, 3)),
+                (tr("Tag vor dem 01. Mai"), date(2029, 4, 30), date(2029, 4, 30)),
```

Comment:
For one-off holidays like this, you can also assign them as part of the `GermanyStaticHolidays` class

i.e. `(tr("Tag vor dem 01. Mai"), date(2029, 4, 30), date(2029, 4, 30)),` here could've been assigned as
```python
    special_hb_school_holidays = {
        # Day before May Day.
        2029: (APR, 30, tr("Tag vor dem 01. Mai")),
    }
```